core/store/migrations/migration1597695690/migrate.go

```diff
@@ -0,0 +1,183 @@
+                error IS NULL AND output IS NOT NULL AND finished_at IS NOT NULL
+                OR
+                output IS NULL AND error IS NOT NULL AND finished_at IS NOT NULL
+            )
+        );
+
+        -- NOTE: This table is large and insert/update heavy so we must be efficient with indexes
+
+        CREATE INDEX idx_pipeline_task_runs ON pipeline_task_runs USING BRIN (created_at);
+        -- CREATE INDEX idx_pipeline_task_runs_finished_at ON pipeline_task_runs WHERE finished_at IS NOT NULL USING BRIN (finished_at);
+
+        -- This query is used in the runner to find unstarted task runs
+        -- CREATE INDEX idx_pipeline_task_runs_unfinished ON pipeline_task_runs (finished_at) WHERE finished_at IS NULL;
+
+        CREATE TABLE offchainreporting_key_bundles (
+            -- NOTE: Key bundle ID is intended to be set by software as sha256 hash of {onchain sig pubkey, offchain sig pubkey, config decryption pubkey}
+            id bytea NOT NULL PRIMARY KEY,
+            CONSTRAINT chk_id_length CHECK (octet_length(id) = 32),
+            encrypted_priv_key_bundle jsonb NOT NULL,
+            created_at timestamptz NOT NULL
+        );
+
+        CREATE INDEX idx_offchainreporting_oracle_specs_unique_key_bundles_created_at ON offchainreporting_key_bundles USING BRIN (created_at);
+
+        CREATE TABLE offchainreporting_oracle_specs (
+            id SERIAL PRIMARY KEY,
+            contract_address bytea NOT NULL,
+            CONSTRAINT chk_contract_address_length CHECK (octet_length(contract_address) = 20),
+
+            -- NOTE THAT THIS LINE HAS BEEN ALTERED TEMPORARILY
+            p2p_peer_id text, -- NOT NULL REFERENCES encrypted_p2p_keys (peer_id),
+
+            p2p_bootstrap_peers jsonb NOT NULL,
+            offchainreporting_key_bundle_id bytea NOT NULL REFERENCES offchainreporting_key_bundles (id),
+            monitoring_endpoint TEXT,
+
+            -- NOTE THAT THIS LINE HAS BEEN ALTERED TEMPORARILY
+            transmitter_address bytea, -- NOT NULL REFERENCES keys (address),
+
+            observation_timeout bigint NOT NULL,
+            blockchain_timeout bigint NOT NULL,
+            contract_config_tracker_poll_interval bigint NOT NULL,
+            contract_config_confirmations INT NOT NULL,
+            created_at timestamptz NOT NULL,
+            updated_at timestamptz NOT NULL
+        );
+
+        -- NOTE: This will be extended with new IDs when we bring directrequest and fluxmonitor under the new jobspawner umbrella
+        -- Only ONE id should ever be present
+        CREATE TABLE jobs (
+            id SERIAL PRIMARY KEY,
+            pipeline_spec_id INT REFERENCES pipeline_specs (id),
+            offchainreporting_oracle_spec_id INT REFERENCES offchainreporting_oracle_specs (id),
+            CONSTRAINT chk_valid CHECK (
+                offchainreporting_oracle_spec_id IS NOT NULL
+            )
+        );
+
+        CREATE FUNCTION deleteConcreteSpecs() RETURNS TRIGGER AS $_$
```

Comment:
whoa, this smells funny to me

Reply:
It's because you can't `ON DELETE CASCADE` in this direction.  If there's a better way, let me know

Reply:
It probably means you designed your schema with foreign keys pointing the wrong way round.